### PR TITLE
Update for Swift 5.10 and strict concurrency

### PR DESCRIPTION
**Pull request by gwynne**
Updates the template for Swift 5.10, removes strict concurrency warnings, and enables them by default. This PR is an extended version of the work started in #124.

Shoutout and thanks to @ameter for the initial work on this!

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -24,17 +24,27 @@ let package = Package(
...
-            .product(name: "Fluent{{fluent.db.module}}Driver", package: "fluent-{{fluent.db.url}}-driver"),{{/fluent}}{{#leaf}}
-            .product(name: "Leaf", package: "leaf"),{{/leaf}}
-        ])
+        .testTarget(
+            name: "AppTests",
+            dependencies: [
+                .target(name: "App"),
+                .product(name: "XCTVapor", package: "vapor"),
+    
+                // Workaround for https://github.com/apple/swift-package-manager/issues/6940
```
Comment (MahdiBM) [inline Package.swift:36]: is this not fixed on 5.10?

**Review by MahdiBM — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (0xTim) [reply Package.swift:36]: I believe so, yes

**Review by 0xTim — CHANGES_REQUESTED** [review]
We can remove the manifest workaround as that works in 5.10

**Comment by 0xTim** [comment]
Closing in favour of #127

**Pull request closed without merging**